packages/plugin-e2e/tests/datasource/queryEditor.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { expect, test } from '../../src';
+import { ProvisionFile } from '../../src/types';
+import { GOOGLE_SHEETS_SPREADSHEETS } from './mocks/resource';
+
+test('should list spreadsheets when clicking on spreadsheet segment', async ({
+  panelEditPage,
+  page,
+  readProvision,
+}) => {
+  const sheetsDataSource = await readProvision<ProvisionFile>({
+    filePath: 'datasources/google-sheets-datasource-jwt.yaml',
+  }).then((provision) => provision.datasources[0]);
+  await panelEditPage.datasource.set(sheetsDataSource.name!);
+  const queryEditorRow = await panelEditPage.getQueryEditorRow('A');
+  await panelEditPage.mockResourceResponse('spreadsheets', GOOGLE_SHEETS_SPREADSHEETS);
+  await queryEditorRow.getByText('Enter SpreadsheetID').click();
+  await expect(page.getByText(GOOGLE_SHEETS_SPREADSHEETS.spreadsheets.sheet1, { exact: true })).toHaveCount(1);
+  await expect(page.getByText(GOOGLE_SHEETS_SPREADSHEETS.spreadsheets.sheet2, { exact: true })).toHaveCount(1);
+});
+
+test('should set correct cache time on query passed to the backend', async ({ panelEditPage, page, readProvision }) => {
+  const sheetsDataSource = await readProvision<ProvisionFile>({
+    filePath: 'datasources/google-sheets-datasource-jwt.yaml',
+  }).then((provision) => provision.datasources[0]);
+  await panelEditPage.datasource.set(sheetsDataSource.name!);
+  const queryEditorRow = await panelEditPage.getQueryEditorRow('A');
+  await panelEditPage.mockResourceResponse('spreadsheets', GOOGLE_SHEETS_SPREADSHEETS);
+  await queryEditorRow.getByText('5m', { exact: true }).click();
+  await page.keyboard.insertText('1h');
+  await page.keyboard.press('Enter');
+
+  const queryReq = panelEditPage.waitForQueryDataRequest((request) =>
+    (request.postData() ?? '').includes('"cacheDurationSeconds":3600')
+  );
+
+  await panelEditPage.refreshPanel();
+  await expect(await queryReq).toBeTruthy();
+});
```

Comment:
What's the "semantic" difference between the tests in `queryEditor.integration.spec.ts` and `queryEditor.spec.ts`? In other words, how do we suggest to organise tests?

Reply:
It's a good question! 

`queryEditor.spec.ts` _just_ tests the query editor. I.e checking the right endpoints are being called when a certain button is clicked and so on. I want to test the query editor in isolation, so any backend endpoints are mocked. 

`queryEditor.integration.spec.ts` is calling the downstream sheets API and fetches data from a readonly spreadsheet. 

Ultimately, it will be up to the plugin author to decide what kind of tests they want to have. Both may be useful. Hitting real downstream APIs can be very useful, but it may come with security concerns and also cause flakiness.